lib/gobstones/compilation_mode.rb

```diff
@@ -0,0 +1,56 @@
+module Gobstones
+  module CompilationMode
+    module Classic
+      class << self
+        def compile_extra(extra)
+          extra
+        end
+
+        def compile_content(content)
+          content
+        end
+      end
+    end
+
+    module GameFramework
+      class << self
+        def compile_extra(extra)
+          [extra, extra_code]
+        end
+
+        def compile_content(content)
+          if blockly_code?(content)
+            xml = Nokogiri::XML(content)
+            xml.root.add_child render_framework_file('program.xml')
+            xml.to_xhtml.gsub(/\n\s*/, '')
+          else
+            <<~GBS
+            #{content}
+
+            #{program_code}
+            GBS
+                .chop
+          end
+        end
+
+        def extra_code
+          render_framework_file 'extra.gbs'
+        end
+
+        def program_code
+          render_framework_file 'program.gbs'
+        end
+
+        private
+
+        def blockly_code?(content)
```

Comment:
Is it possible this code is already duplicated somewhere else in this repo?

Reply:
I don't think so. We have something similar _elsewhere_

Reply:
Before this feature, the runner didn't care about the format of the solution: it just redirected it to the `gobstones-cli`. There is something similar there:

```js
var isBlocklyCode = _.startsWith(code, "<xml");
```

:shrug: